### PR TITLE
Deprecate HttpSensorAsync

### PR DESCRIPTION
**Pull request by vatsrahul1001**
This PR deprecates HttpSensorAsync by proxying them to their Airflow OSS provider's.

closes:https://github.com/astronomer/astronomer-providers/issues/1420

### CODE REVIEW

File: astronomer/providers/http/hooks/http.py [hunk 1]
```diff
@@ -43,6 +35,14 @@ def __init__(
         *,
         keep_response: bool = False,
     ) -> None:
+        warnings.warn(
+            (
+                "This class is deprecated. "
+                "Use `airflow.providers.dbt.cloud.hooks.dbt.DbtCloudHook` instead."
```
Comment (pankajkoti) [inline astronomer/providers/http/hooks/http.py:41]: We need to correct it to Http 🙂

**Comment by pankajastro** [comment]
Note: bump http provider too https://github.com/astronomer/astronomer-providers/blob/main/setup.cfg

File: astronomer/providers/sftp/triggers/sftp.py
Comment (pankajastro) [inline astronomer/providers/sftp/triggers/sftp.py:1]: Why we are deleting this file?

Comment (vatsrahul1001) [reply astronomer/providers/sftp/triggers/sftp.py:1]: It should not, I will fix this

**Comment by vatsrahul1001** [comment]
Ran example DAG to verify DAG is doing in deferred state

image

**Review by pankajkoti — APPROVED** [review]

**Pull request merged**